kitsune/sumo/handlers.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from dataclasses import dataclass, field
+from typing import Generic, TypeVar
+
+T = TypeVar("T")
+
+
+class AbstractHandler(ABC):
+    """Abstract base class for all handlers."""
+
+    @abstractmethod
+    def handle(self) -> None:
+        pass
+
+
+@dataclass
+class AbstractChain(Generic[T]):
+    """Abstract base class for a chain of handlers."""
+
+    _handlers: list[T] = field(default_factory=list)
+
+    @property
+    def handlers(self) -> Sequence[T]:
+        return self._handlers
+
+    def add_handler(self, handler: T) -> None:
+        """Add a handler to the chain."""
+        self._handlers.append(handler)
```

Comment:
Within this class, why not use `AbstractHandler` instead of `T`? I can't see that it would ever be anything other than an `AbstractHandler`.

Reply:
True but the problem lies in the signature of `AccountHandler`. This defines an extra method `handle_account`. I did that in order to keep the `AbstractHandler` generic for future use and create a specific handler for accounts. Because of this, mypy will raise an error for the `AAQChain(AbstractChain[AccountHandler])` definition if the `AbstractChain` is specific to `AbstractHandler`